File: z-clients/javascript/src/models/msgPublishIn.ts
// this file is @generated
import {
    type MsgIn,
    MsgInSerializer,
} from './msgIn';

export interface MsgPublishIn {
    namespace?: string | null;
    msgs: MsgIn[];
    idempotencyKey?: string | null;
}

export interface MsgPublishIn_ {
    namespace?: string | null;
    topic: string;
    msgs: MsgIn[];
    idempotencyKey?: string | null;
}

export const MsgPublishInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgPublishIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            msgs: object['msgs'].map((item: MsgIn) => MsgInSerializer._fromJsonObject(item)),
            idempotencyKey: object['idempotency_key'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgPublishIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'msgs': self.msgs.map((item) => MsgInSerializer._toJsonObject(item)),
            'idempotency_key': self.idempotencyKey,
        };
    }
}

File: z-clients/rust/src/models/msg_publish_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::msg_in::MsgIn;

#[derive(Clone, Debug, Deserialize)]
pub struct MsgPublishIn {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub namespace: Option<String>,

    pub msgs: Vec<MsgIn>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub idempotency_key: Option<String>,
}

impl MsgPublishIn {
    pub fn new(msgs: Vec<MsgIn>) -> Self {
        Self {
            namespace: None,
            msgs,
            idempotency_key: None,
        }
    }

    pub fn with_namespace(mut self, value: impl Into<Option<String>>) -> Self {
        self.namespace = value.into();
        self
    }

    pub fn with_idempotency_key(mut self, value: impl Into<Option<String>>) -> Self {
        self.idempotency_key = value.into();
        self
    }
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub(crate) struct MsgPublishIn_ {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub namespace: Option<String>,

    pub topic: String,

    pub msgs: Vec<MsgIn>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub idempotency_key: Option<String>,
}
